common/test/acceptance/tests/discussion/test_discussion.py

```diff
@@ -447,6 +456,106 @@ def edit_response(self, page, response_id):
+        page.start_response_edit(response_id)
+        url = "http://www.example.com/somewhere"
+        description = "somewhere on example.com"
+        page.set_response_editor_value(response_id, "")
+        page.add_content_via_editor_button("link", response_id, url, description)
+        page.submit_response_edit(response_id, description)
+        expected_response_html = (
+            '<p><a href="http://www.example.com/somewhere">'
+            'somewhere on example.com</a></p>'
+        )
+        self._confirm_response_html(page, response_id, expected_response_html)
+
+    def edit_response_add_link_errors(self, page, response_id):
+        """
+        Replacing the current content with a link via the "add link"
+        button in the editor.
+        """
+        self.assertTrue(page.is_response_editable(response_id))
+        page.start_response_edit(response_id)
+        page.set_response_editor_value(response_id, "")
+        page.add_content_via_editor_button("link", response_id, '', '')
+        expected_response_html = (
+            '<p><a href="http://www.example.com/somewhere">'
+            'somewhere on example.com</a></p>'
+        )
+        self._confirm_response_html(page, response_id, expected_response_html)
+
+    def edit_response_add_image(self, page, response_id):
+        """
+        Replacing the current content with an image via the "add image"
+        button in the editor.
+        """
+        self.assertTrue(page.is_response_editable(response_id))
+        page.start_response_edit(response_id)
+        url = "http://www.example.com/something.png"
+        description = "image from example.com"
+        page.set_response_editor_value(response_id, "")
+        page.add_content_via_editor_button("image", response_id, url, description)
+        page.submit_response_edit(response_id, '')
+        expected_response_html = (
+            '<p><img src="http://www.example.com/something.png"'
+            ' alt="image from example.com" title=""></p>'
+        )
+        self._confirm_response_html(page, response_id, expected_response_html)
+
+    def verify_link_editor_error_messages_shown(self, page):
+        """
+        Confirm that the error messages are displayed in the editor.
+        """
+        def errors_visible():
+            """
+            Returns True if both errors are visible, False otherwise.
+            """
+            return (
+                page.q(css="#new-url-input-field-message.has-error").visible and
+                page.q(css="#new-url-desc-input-field-message.has-error").visible
+            )
+
+        page.wait_for(errors_visible, "Form errors should be visible.")
+
+    def test_edit_response_add_link(self):
+        self.setup_user()
+        self.setup_view()
+        page = self.create_single_thread_page("response_edit_test_thread")
+        page.visit()
+        self.edit_response_add_link(page, "response_self_author")
```

Comment:
IMO I don't like tests that don't have any explicit checks. I think it is okay to have helper methods to set up the correct state of the page, but then I prefer to see the tests inline. It makes it easier to 'read' the test to see what it is doing. In this case, the helper function is only called in one place (here) so it isn't helping reuse either. I might refactor this helper into a method on the page object that sets the state, and then I would have the assertions inline here.